### PR TITLE
Search Google Books

### PR DESCRIPTION
**Pull request by Ali-Parandeh**
## Description

Search Google Books

## Type of change

- New extension

## Screencast

https://user-images.githubusercontent.com/43654026/150417366-573cc573-c772-4eff-87c2-1b125560c53e.mov

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

File: extensions/google-books/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Search Google Books from Raycast
+
+Plugin for Raycast to search [Google Books](https://books.google.co.uk/).
```
Comment (pernielsentikaer) [inline extensions/google-books/README.md:3]:
```suggestion
Extension for Raycast to search [Google Books](https://books.google.co.uk/).
```

File: extensions/google-books/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+
+## Installing
+
+Make sure RayCast is installed and go to https://github.com/Ali-Parandeh/raycast-google-books-extension
```
Comment (pernielsentikaer) [inline extensions/google-books/README.md:8]:
```suggestion
Make sure Raycast is installed and go to https://github.com/Ali-Parandeh/raycast-google-books-extension
```

File: extensions/google-books/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                  title={item.volumeInfo.title}
+                  subtitle={item?.volumeInfo?.authors ? item.volumeInfo.authors[0] : "Various Authors"}
+                  accessoryTitle={
+                    (item?.volumeInfo?.averageRating ? "  ⭐  " + item.volumeInfo.averageRating + "/5" : "  ⭐  N/A") +
+                    (item.volumeInfo?.pageCount ? "  📄  " + item.volumeInfo.pageCount : "  📄  N/A")
+                  }
+                  actions={
+                    <ActionPanel>
+                      <PushAction
+                        title="Show Book Details"
```
Comment (pernielsentikaer) [inline extensions/google-books/src/index.tsx:60]:
```suggestion
                        title="Show Book Details"
                        icon={Icon.List}
```
Missing icon

File: extensions/google-books/package.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  "name": "google-books",
+  "title": "Google Books",
+  "description": "Search Books with Google Books",
+  "icon": "google-books.png",
+  "author": "aparandeh",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Search Google Books",
```
Comment (pernielsentikaer) [inline extensions/google-books/package.json:12]:
```suggestion
      "title": "Search Books",
      "subtitle": "Google Books",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @Ali-Parandeh 👋 

Awesome extension 💪 I left some feedback for you to address 🚀 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by Ali-Parandeh** [comment]
@pernielsentikaer Thanks for the improvements! 💪 All committed :)

File: extensions/google-books/src/index.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+                        icon={Icon.List}
+                        target={
+                          <Detail
+                            actions={
+                              <ActionPanel>
+                                <OpenInBrowserAction icon={Icon.Globe} url={item.volumeInfo.infoLink} />
+                                <CopyToClipboardAction icon={Icon.Clipboard} content={item.volumeInfo.infoLink} />
```
Comment (pernielsentikaer) [inline extensions/google-books/src/index.tsx:67]:
```suggestion
                                <CopyToClipboardAction title="Copy URL to Clipboard" icon={Icon.Clipboard} content={item.volumeInfo.infoLink} />
```

File: extensions/google-books/src/index.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                            }
+                            navigationTitle={item.volumeInfo.title}
+                            markdown={`${item?.volumeInfo?.subtitle ?? "" + "\n\n"} ${
+                              item?.volumeInfo?.description ?? ""
+                            }`}
+                          />
+                        }
+                      />
+                      <OpenInBrowserAction icon={Icon.Globe} url={item.volumeInfo.infoLink} />
+                      <CopyToClipboardAction icon={Icon.Clipboard} content={item.volumeInfo.infoLink} />
```
Comment (pernielsentikaer) [inline extensions/google-books/src/index.tsx:78]:
```suggestion
                      <CopyToClipboardAction icon={Icon.Clipboard} title="Copy URL to Clipboard" content={item.volumeInfo.infoLink} />
```

File: extensions/google-books/src/index.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+                        target={
+                          <Detail
+                            actions={
+                              <ActionPanel>
+                                <OpenInBrowserAction icon={Icon.Globe} url={item.volumeInfo.infoLink} />
+                                <CopyToClipboardAction icon={Icon.Clipboard} content={item.volumeInfo.infoLink} />
+                              </ActionPanel>
+                            }
+                            navigationTitle={item.volumeInfo.title}
+                            markdown={`${item?.volumeInfo?.subtitle ?? "" + "\n\n"} ${
```
Comment (pernielsentikaer) [inline extensions/google-books/src/index.tsx:71]:
```suggestion
                            markdown={`## ${item.volumeInfo.title} ${item?.volumeInfo?.subtitle ?? "" + "\n\n"} ${
```
Just a suggestion to add the title as header

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @Ali-Parandeh 👋 

Found three more, didn't see the copy action was for the link 😄 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by Ali-Parandeh** [comment]
@pernielsentikaer  New feature!

Added Thumbnails to book lists, and properly formatted book details page

Screenshot 2022-01-21 at 12 55 41

Screenshot 2022-01-21 at 12 55 56

Screenshot 2022-01-21 at 12 56 01

**Comment by pernielsentikaer** [comment]
Much better @Ali-Parandeh – Very awesome! 

You forgot to add `icon={Icon.List}` (or another icon to `Show Book Details` action item, can you add that? After that, we're ready 😄 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by Ali-Parandeh** [comment]
All Done @pernielsentikaer :)

**Review by pernielsentikaer — APPROVED** [review]
Awesome @Ali-Parandeh 🚀 

Thanks for an awesome contribution 💪 

I'm going to accept it and then will my colleague do a final review before it's finally merged.

**Review by mathieudutour — APPROVED** [review]

**Pull request merged**